source/GameInterface/Services/MobileParties/Patches/UnitRecruitPatch.cs

```diff
@@ -0,0 +1,123 @@
+﻿using Common;
+using Common.Extensions;
+using Common.Logging;
+using Common.Messaging;
+using Common.Util;
+using GameInterface.Policies;
+using GameInterface.Services.MobileParties.Extensions;
+using GameInterface.Services.MobileParties.Handlers;
+using GameInterface.Services.MobileParties.Messages;
+using HarmonyLib;
+using Serilog;
+using Serilog.Core;
+using System;
+using System.Reflection;
+using TaleWorlds.CampaignSystem;
+using TaleWorlds.CampaignSystem.Actions;
+using TaleWorlds.CampaignSystem.Party;
+using TaleWorlds.CampaignSystem.Roster;
+using TaleWorlds.CampaignSystem.ViewModelCollection.CharacterDeveloper;
+
+namespace GameInterface.Services.MobileParties.Patches
+{
+    [HarmonyPatch(typeof(TroopRoster))]
+    public class UnitRecruitPatch
+    {
+        private static PropertyInfo TroopRoster_OwnerParty => typeof(TroopRoster).GetProperty("OwnerParty", BindingFlags.Instance | BindingFlags.NonPublic);
+
+        private static readonly MethodInfo TroopRoster_AddNewElement = typeof(TroopRoster).GetMethod("AddNewElement", BindingFlags.NonPublic | BindingFlags.Instance);
```

Comment:
Can you convert this to a compiled method (for example)[https://github.com/Bannerlord-Coop-Team/BannerlordCoop/blob/development/source/GameInterface/Services/MobilePartyAIs/Patches/PartyBehaviorPatch.cs#L144]